File: BackEnd/java/ZaloMini/src/main/java/com/devmam/zalomini/entities/UserSession.java
package com.devmam.zalomini.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.Instant;
import java.util.Date;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity(name = "UserSession")
@Table(name = "user_sessions")
public class UserSession {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "device_id")
    private String deviceId;

    @Column(name = "device_type", length = 50)
    private String deviceType;

    @Lob
    @Column(name = "access_token", nullable = false)
    private String accessToken;

    @Lob
    @Column(name = "refresh_token")
    private String refreshToken;

    @Column(name = "expires_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date expiresAt;

    @ColumnDefault("1")
    @Column(name = "is_active")
    private Boolean isActive;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "created_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

}

File: BackEnd/Zalo_mini/Zalo_mini/Models/user_session.cs
using System;
using System.Collections.Generic;

namespace Zalo_mini.Models
{
    public partial class user_session
    {
        public long id { get; set; }
        public long user_id { get; set; }
        public string? device_id { get; set; }
        public string? device_type { get; set; }
        public string access_token { get; set; } = null!;
        public string? refresh_token { get; set; }
        public DateTime? expires_at { get; set; }
        public bool? is_active { get; set; }
        public DateTime? created_at { get; set; }

        public virtual user user { get; set; } = null!;
    }
}
